### PR TITLE
Add FrameworkStatusReportService for MigrationInvoker

### PR DESCRIPTION
**Pull request by BurningCN**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java [hunk 1]
```diff
@@ -425,10 +428,10 @@ protected void refreshServiceDiscoveryInvoker(CountDownLatch latch) {
         }
         setListener(serviceDiscoveryInvoker, () -> {
             latch.countDown();
-            //TODO FrameworkStatusReporter
-//            FrameworkStatusReporter.reportConsumptionStatus(
-//                createConsumptionReport(consumerUrl.getServiceInterface(), consumerUrl.getVersion(), consumerUrl.getGroup(), "app")
-//            );
+            if (reportService.hasReporter()) {
+                reportService.reportMigrationStepStatus(
```
Comment (AlbumenJ) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java:432]: change to `reportService.reportConsumptionStatus`

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java [hunk 2]
```diff
@@ -449,10 +452,10 @@ protected void refreshInterfaceInvoker(CountDownLatch latch) {
         }
         setListener(invoker, () -> {
             latch.countDown();
-            //TODO FrameworkStatusReporter
-//            FrameworkStatusReporter.reportConsumptionStatus(
-//                createConsumptionReport(consumerUrl.getServiceInterface(), consumerUrl.getVersion(), consumerUrl.getGroup(), "interface")
-//            );
+            if (reportService.hasReporter()) {
+                reportService.reportMigrationStepStatus(
```
Comment (AlbumenJ) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java:456]: change to `reportService.reportConsumptionStatus`

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java (hunk 1 above)
Comment (BurningCN) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java]: done

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java (hunk 2 above)
Comment (BurningCN) [reply dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/migration/MigrationInvoker.java]: done

**Pull request merged**